### PR TITLE
re-scale sample weights to 1 mean

### PR DESCRIPTION
**Pull request by alexander-rakhlin**
Small sample weights result in diminished gradients, and estimator never learns. This is a typical outcome of setting weights to 1/number_of_samples like in sklearn AdaBoostClassifier

### CODE REVIEW

**Comment by Laurae2** [comment]
cf your PR at https://github.com/Microsoft/LightGBM/pull/407 for more context as to why it is a bad idea.
